dblookupext/esdtSupply/logsProcessor.go

```diff
@@ -0,0 +1,132 @@
+package esdtSupply
+
+import (
+	"bytes"
+	"math/big"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go-core/data"
+	"github.com/ElrondNetwork/elrond-go-core/data/transaction"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+type logsProcessor struct {
+	marshalizer        marshal.Marshalizer
+	suppliesStorer     storage.Storer
+	fungibleOperations map[string]struct{}
+}
+
+func newLogsProcessor(
+	marshalizer marshal.Marshalizer,
+	suppliesStorer storage.Storer,
+) *logsProcessor {
+	return &logsProcessor{
+		marshalizer:    marshalizer,
+		suppliesStorer: suppliesStorer,
+		fungibleOperations: map[string]struct{}{
+			core.BuiltInFunctionESDTLocalBurn:      {},
+			core.BuiltInFunctionESDTLocalMint:      {},
+			core.BuiltInFunctionESDTNFTCreate:      {},
+			core.BuiltInFunctionESDTNFTAddQuantity: {},
+			core.BuiltInFunctionESDTNFTBurn:        {},
+		},
+	}
+}
+
+func (lp *logsProcessor) processLogs(logs map[string]data.LogHandler, isRevert bool) error {
```

Comment:
bool param. This function clearly does 2 things. Should split in 2.

Reply:
If I split it into two functions will have a lot of duplicated code.
This function does one thing ---> process a log ---> IF is revert in case of Burn ESDT or NFT will add to supply instead of decrease.

Reply:
could have used a "multiplier" factor (1 or -1) instead of bool